crates/ntx-builder/src/state/mod.rs

```diff
@@ -30,14 +31,33 @@ mod account;
 pub struct TransactionCandidate {
     /// The current inflight state of the account.
     pub account: Account,
+
     /// A set of notes addressed to this network account.
     pub notes: Vec<NetworkNote>,
+
+    /// The latest locally committed block header.
+    ///
+    /// This should be used as the reference block during transaction execution. It is guaranteed
+    /// to be tracked by `chain_mmr`.
+    pub chain_tip: BlockHeader,
+
+    /// The chain MMR, which includes the `chain_tip`'s block.
+    pub chain_mmr: PartialBlockchain,
 }
 
 /// Holds the state of the network transaction builder.
 ///
 /// It tracks inflight transactions, and their impact on network-related state.
 pub struct State {
+    /// The latest committed block header.
+    chain_tip: BlockHeader,
+
+    /// The chain MMR including the latest block header.
+    chain_mmr: PartialBlockchain,
+
+    /// The number of blocks in the chain MMR after pruning.
+    chain_mmr_block_count: u32,
```

Comment:
I believe we need to track this manually because there is no way to get block count from the mmr, only total chain length (latest block number) AFAICT.